### ISSUE TITLE
Page routing for options-general.php doesn't work as expected

### ISSUE THREAD

**Issue by djgadd** [themosis/framework#648]
```
$this->app['page']->make('test', 'Test Settings')
    ->setParent('options-general.php')
    ->set();

$page->route('/', function () {
    dd('hello world');
});
```

Doesn't behave as expected, instead of setting 'hello world' I see the default Themosis settings page (with a link to the Page API documentation.)

I _think_ the issue is one of: -

1. Page::findParentHook not returning 'settings'; or if that's not the intention of that method
2. Page::registerRouteActions not adding a `load-settings_page_'.$page->getSlug()` action.

Manually adding the `load-settings-...` action gives me the result I was expecting.

**Comment by jlambe** [comment]
Hi @djgadd, I'm going to test this. Probably there is a missing hook name missing as you mention in your solution.

I mark this for minor release. Would you like to submit a PR against the `2.0` branch of the project ?

**Comment by djgadd** [comment]
@jlambe I'll see if I can get one submitted this weekend. What's your preference, should this be handled in the `findParentHook` method, or just an additional hook in `registerRouteActions`?

**Comment by jlambe** [comment]
I think the best solution would be to check the `findParentHook` and return the `settings` parent hook  name for this scenario.

Not sure about other core parent pages like appearance, ... perhaps there is a RegExp to look for on the user given parent string (ex: `options-general.php`, `themes.php`, ...) or if we just need to write conditions for all core parent pages.

**Comment by djgadd** [comment]
Leave it with me, I'll write some tests and see what I can do 👍

**Comment by jlambe** [comment]
@djgadd I'm going to move your issue for next minor release `2.0.2` as I need to move forward for other tiny issues and release an updated version. This will give you more time to look at this.

**Comment by besrabasant** [comment]
I am also facing the same issue with hooks. I have added a Custom Admin Page and Sub Page. The hooks on `registerRouteActions` method gets called on The Admin Parent Page  but not on the Sub Page.

**Comment by jlambe** [comment]
Issue has been fixed within the PR #657.
